core/src/main/java/feign/SynchronousMethodHandler.java

```diff
@@ -37,13 +34,13 @@
   static class Factory {
 
     private final Client client;
-    private final Provider<Retryer> retryer;
-    private final Set<RequestInterceptor> requestInterceptors;
+    private final Retryer retryer;
+    private final List<RequestInterceptor> requestInterceptors;
     private final Logger logger;
-    private final Provider<Logger.Level> logLevel;
+    private final Logger.Level logLevel;
```

Comment:
no longer modifiable at runtime, but I'm not sure anyone ever did. Ex. the Builder never exposed a provider for logLevel. Let's add support for dynamic switching only if asked.